06_puppy/alextmz/mapstore.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+)
+
+type MapStore map[int]Puppy
+
+func NewmapStore() MapStore {
+	a := MapStore{}
+	return a
+}
+
+func (m MapStore) CreatePuppy(p *Puppy) error {
+	switch {
+	case p == nil:
+		return fmt.Errorf("error, puppy pointer is nil")
+	case p.ID == 0:
+		p.ID = len(m) + 1
+		m[p.ID] = *p
+		return nil
+	default:
+		return fmt.Errorf("trying to create a puppy that already has an ID %d", p.ID)
+	}
+}
+
+func (m MapStore) ReadPuppy(id int) (Puppy, error) {
+	if id < 0 {
+		return Puppy{}, fmt.Errorf("invalid puppy id %d", id)
+	}
```

Comment:
i wouldn't test for this at all separately as it should be covered by the `ok` check, but if you insist, maybe just work with `uint`?

Reply:
I'm ok with taking those off.